backend/packages/Upgrade/src/api/Algorithms.ts

```diff
@@ -0,0 +1,200 @@
+        tempFactorArray.splice(randomConditionIndex, 1);
+      }
+    }
+  }
+
+  const randomRoundRobinAssignData = {
+    site: assignedData.site,
+    target: assignedData.target,
+    assignedCondition: randomRoundRobinConditionArray,
+    assignedFactor: experiment.type === EXPERIMENT_TYPE.FACTORIAL ? assignedFactorsArray : null,
+  };
+
+  // rotate elements in assigned condition array based on number of monitored decision point
+  return rotateElements(randomRoundRobinAssignData, monitoredDecisionPointLogsLength);
+}
+
+export function rotateElements(
+  assignedData: IExperimentAssignmentv5,
+  monitoredDecisionPointLogsLength: number
+): IExperimentAssignmentv5 {
+  if (monitoredDecisionPointLogsLength > 0 && assignedData.assignedCondition.length >= 2) {
+    const totalloopIteration = monitoredDecisionPointLogsLength % assignedData.assignedCondition.length;
+
+    for (let i = 0; i < totalloopIteration; i++) {
+      const assignedCondition = assignedData.assignedCondition.shift();
+      assignedData.assignedCondition.push(assignedCondition);
+
+      if (assignedData.assignedFactor) {
+        const assignedFactor = assignedData.assignedFactor.shift();
+        assignedData.assignedFactor.push(assignedFactor);
+      }
+    }
+  }
+  return assignedData;
+}
+
+function convertToAssignedCondition(
+  experiment: Experiment,
+  conditionPayloads: ConditionPayloadDTO[],
+  site: string,
+  target: string,
+  factors: FactorDTO[]
+): IExperimentAssignmentv5 {
+  const assignedConditionArray: AssignedCondition[] = [];
+  const assignedFactorsArray: Record<string, { level: string; payload: IPayload }>[] = [];
+
+  experiment.conditions.forEach((condition) => {
+    let conditionPayload: ConditionPayloadDTO = null;
+    let factorialObject;
+
+    if (experiment.type === EXPERIMENT_TYPE.FACTORIAL) {
+      // returns factorial alias condition or assigned condition
+      conditionPayload = conditionPayloads.find((cP) => cP.parentCondition.id === condition.id);
+      factorialObject = getAssignedFactor(condition, factors);
+    } else {
+      // checking alias condition for simple experiment
+      conditionPayload = conditionPayloads.find(
+        (cP) =>
+          cP.parentCondition.id === condition.id && cP.decisionPoint.site === site && cP.decisionPoint.target === target
+      );
+    }
+
+    const assignedCondition = {
+      conditionCode: condition.conditionCode,
+      payload: conditionPayload?.payload,
+      experimentId: experiment.id,
+      id: condition.id,
+    };
+    assignedConditionArray.push(assignedCondition);
+    factorialObject ? assignedFactorsArray.push(factorialObject) : null;
+  });
+
+  return {
+    site: site,
+    target: target,
+    assignedCondition: assignedConditionArray,
+    assignedFactor: experiment.type === EXPERIMENT_TYPE.FACTORIAL ? assignedFactorsArray : null,
+  };
+}
+
+function getAssignedFactor(conditionAssigned: ExperimentCondition, factors: FactorDTO[]): object {
```

Comment:
Change the return type to proper typing `object` is too generic

Reply:
code updated